src/index.js

```diff
@@ -24,21 +28,25 @@ function WalletWatcher({ children }) {
       const signer = provider ? await provider.getSigner() : undefined;
       const walletAddress = accounts[0] ? accounts[0].toLowerCase() : undefined;
       const token = restoreToken({ walletAddress });
+      const chainId = await window.ethereum.request({ method: 'eth_chainId' });
 
       updateSynthetix({
         walletAddress,
         token,
         provider,
         signer,
+        chainId,
       });
     }
 
     window.ethereum.on('accountsChanged', onAccountsChanged);
+    window.ethereum.on('chainChanged', handleChainChanged);
```

Comment:
```suggestion
    const handleChainChanged = (chainId) => updateSynthetix({ chainId });
    
    window.ethereum.on('chainChanged', handleChainChanged);
```

Dont think callback is necessary. Both callbacks for accounts changed and chain changed can live right here in place and not participate in react event loop

Reply:
Of course! Good for you for noticing that! Moved the callback.